dts/bindings/serial/espressif,esp32-uart.yaml

```diff
@@ -30,3 +30,8 @@ properties:
       type: int
       description: CTS pin
       required: false
+
+    peripheral-id:
+      type: int
+      description: Peripheral ID
+      required: true
```

Comment:
I don't see this being used anywhere in the driver; what's going on?

Reply:
In fact, this is used in HAL:  
```c
void IRAM_ATTR esp_restart_noos_dig(void)
{
	/* make sure all the panic handler output is sent from UART FIFO */
	if (CONFIG_SERIAL && DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_console), okay)) {
		esp_rom_uart_tx_wait_idle(DT_PROP(DT_CHOSEN(zephyr_console), peripheral_id));
	}

	/* switch to XTAL (otherwise we will keep running from the PLL) */
	rtc_clk_cpu_freq_set_xtal();

#if CONFIG_IDF_TARGET_ESP32
	esp_cpu_unstall(PRO_CPU_NUM);
#endif
	/* reset the digital part */
	SET_PERI_REG_MASK(RTC_CNTL_OPTIONS0_REG, RTC_CNTL_SW_SYS_RST);
	while (true) {
		;
	}
}
```

Reply:
Oh, ok. Thanks for clarifying.

However, isn't glue code supposed to be put into the zephyr repository, not the HAL repository, these days? I seem to recall a process working group decision along those lines.

Reply:
@mbolivar you are correct, that code should not sit in there. I removed from this MR. Will push in another one later.  PTAL.

Reply:
> Will push in another one later. PTAL.

OK, I think there's nothing left for me to review here, then. When the new PR comes in I'll take a look